c7n/resources/awslambda.py

```diff
@@ -655,3 +655,47 @@ def format_resource(self, r):
         payload.update(self.filter_empty(
             select_keys(r, ['Version', 'CreatedDate', 'CompatibleRuntimes'])))
         return envelope
+
+
+@AWSLambda.filter_registry.register('lambda-edge')
+class LambdaVersionEdgeFilter(Filter):
+    """
+    Filter for lambda@edge functions
+
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+                - name: lambda-edge-filter
+                  resource: lambda
+                  filters:
+                    - type: lambda-edge
+                      state: True
+    """
+    permissions = ('cloudfront:ListDistributions',)
+
+    schema = type_schema('lambda-edge',
+        **{'state': {'type': 'boolean'}})
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('cloudfront')
```

Comment:
just using the cloudfront resource manager will get us the same and get cache reuse.